openedx/features/redhouse_features/management/commands/sync_users_data.py

```diff
@@ -0,0 +1,137 @@
+
+        sub_org = self.get_sub_organization(sub_organization)
+
+        linked_users_count = len(
+            [
+                1 for user in users if self.create_and_link_edly_profiles(user, sub_org)
+            ]
+        )
+
+        course_creator_count = len([
+            1 for user in staff_users if self.create_course_creator(user)
+        ])
+        course_access_roles_count = len([
+            1 for user in staff_users if self.create_course_access_roles(user, organization)
+        ])
+
+        map(self.remove_staff_access, staff_users)
+
+        log.info('Edly user profiles are successfully created for %d user(s)', linked_users_count)
+        log.info('Course creator objects are successfully created for %d user(s)', course_creator_count)
+        log.info('Course access roles are successfully created for %d user(s)', course_access_roles_count)
+        log.info('Staff access is removed for %d user(s)', course_access_roles_count)
+
+    def get_sub_organization(self, sub_org_slug):
+        try:
+            return EdlySubOrganization.objects.get(slug=sub_org_slug)
+        except EdlySubOrganization.DoesNotExist:
+            log.error('Edly sub organization %s does not exist', sub_org_slug)
+            exit(1)
+
+    def create_and_link_edly_profiles(self, user, sub_org):
+        try:
+            edly_profile = EdlyUserProfile.objects.get(user=user)
+            log.info(
+                'Found existing edly user profile %s linked with sub org %s',
+                user.username,
+                sub_org
+            )
+            return
+        except EdlyUserProfile.DoesNotExist:
+            edly_profile = EdlyUserProfile.objects.create(user=user)
+            edly_profile.edly_sub_organizations.add(sub_org)
+            edly_profile.save()
+            log.info(
+                'Created edly user profile for user %s and linked with sub org %s',
+                user.username,
+                sub_org
+            )
+            return edly_profile
+
+    def create_course_creator(self, user):
+        try:
+            course_creator = CourseCreator.objects.get(user=user)
+            log.info(
+                'Found existing CourseCreator for user %s with state %s',
+                user.username,
+                course_creator.state
+            )
+            return
+        except CourseCreator.DoesNotExist:
+            course_creator = CourseCreator.objects.create(
+                user=user,
+                state=CourseCreator.GRANTED,
+                note='Auto-generated using the management command.'
+            )
+
+            log.info(
+                'Created CourseCreator for user %s with status %s',
+                user.username,
+                CourseCreator.GRANTED
+            )
+
+            return course_creator
+
+    def create_course_access_roles(self, user, org):
+        try:
+            course_access_role = CourseAccessRole.objects.get(user=user)
```

Comment:
We want to check specific role, so this query will not work here at all.

Reply:
This is just a first time fix for the existing users. This command is not intended to be used in future therefore its very unlikely that there is an existing access role for a user.
And even if there is (in this case), we don't want to override/change that existing one.

Reply:
I didn't get it, however leaving it onto you.